restful_api/controllers/token.py

```diff
@@ -0,0 +1,98 @@
+# Part of odoo. See LICENSE file for full copyright and licensing details.
+import json
+import logging
+from urllib import response
+import werkzeug.wrappers
+import hashlib
+import os
+from base64 import b64encode
+from os import urandom
+
+from odoo import http
+from odoo.addons.restful_api.common import invalid_response, valid_response
+from odoo.http import request
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+import datetime
+
+_logger = logging.getLogger(__name__)
+
+def default(o):
+    if isinstance(o, (datetime.date, datetime.datetime)):
+        return o.isoformat()
+
+expires_in = "restful_api.access_token_expires_in"
+
+#--------------------------------- Access_Key Login url ----------------------------------------#
+
+class AccessToken(http.Controller):
+    """."""
+
+    def __init__(self):
+
+        self._token = request.env["api.access_token"]
+        self._expires_in = request.env.ref(expires_in).sudo().value
+
+    @http.route("/api/login", methods=["POST","OPTIONS"], type="json", auth="none", csrf=False, cors='*')
+    def token(self, **post):
+        try:
+            input_user_name = post.get("data").get("username"),
+            input_password = post.get("data").get("password")
+            if input_user_name[0] != None:
+                res_usersss = request.env['res.users'].sudo().search([
+                    ('active','=',True),
+                    ('login','=',input_user_name)])
+                
+                
+                if res_usersss:
+                    request.env.company_id = 1  ###set company_id based logic users 
+                    exp = datetime.datetime.utcnow() + datetime.timedelta(days=30)
```

Comment:
Here it seems token is set to expire after 30 days. Earlier it was 1 year. Can you explain what would happen if this was set to 24 hours, or less?

Reply:
The token stored in the session for further use, the token valid from generated date to after 30 days, In-case for each request logic hits the login api and request with new token means this logic not needed, better to be 24 hours or lesser than that ,I set as 24 hours.